### ISSUE TITLE
diamond --threads

### ISSUE THREAD

**Issue by alephreish** [AlexanderLabWHOI/EUKulele#50]
EUKulele (testing on 2.0.0) doesn't appear to be setting the `--threads` argument for diamond, irrespective of whether `--CPUs` argument was set or not. By default, diamond uses all available CPU power, so the user of EUKulele has no control over how many cores are actually used.

**Comment by akrinos** [comment]
An update on this: adding the feature of `--CPUs` being leveraged for all DIAMOND calls, which was not true previously. Will push an update this week of this + dependency version fix

**Comment by akrinos** [comment]
@alephreish Were you ever able to install a newer version of EUKulele to test this out? thanks!

**Comment by hrzolix** [comment]
I was testing EUKulele 2.0.6 with tutorial on official page. I put CPUs: 6 in the curr_config.yaml, even tried with command line --CPUs 6, still diamond uses all of the cpus available.

**Comment by akrinos** [comment]
@hrzolix let me know how this turned out and feel free to reopen - at present I am getting the right CPU behavior locally

**Comment by alephreish** [comment]
@akrinos We're running EUKulele v. 2.0.9. It still looks like diamond is trying to use all of the cores:

````
# EUKulele call:
EUKulele --sample_dir my_sample_dir -m mets --protein_extension .pep --out_dir results --reference_dir databases --database eukprot --CPUs 20
#
# diamond process:
diamond blastp --db databases/eukprot/diamond/reference.pep.dmnd -q my_sample_dir/longest_orfs.pep -o results/mets_full/diamond/longest_orfs.diamond.out --outfmt 6 -p 64 -k 100 -e 1e-05 -b3.0
````

**Comment by akrinos** [comment]
Hey @alephreish ! Sorry about that. You are right that this is not being passed properly in 2.0.9 (and is not in any current version for `BLAST`). Shortly I will be pushing a fix to PyPI as version 2.1.0 that hopefully you can test. Thanks for getting back and for using `EUKulele`!

**Comment by akrinos** [comment]
You can try the pip install here now @alephreish - https://pypi.org/project/EUKulele/2.1.0/
